### PR TITLE
fix zombies and xenos being able to pry bolted doors

### LINKED ISSUE #20275: Zombies can pry bolted doors
## Description
Zombies can pry bolted doors, which are then bolted open after they pry them. 

**Reproduction**
Pry a bolted door as a zombie.

### PR DESCRIPTION
**Pull request by MisterMecky**
## About the PR

Makes zombies and xenos unable to pry open doors that are bolted.

## Why / Balance

Assumed to be a bug.
Resolves #20275

## Technical details

The TryPryDoor() in alt interact was set to force = true, which meant it never checked if the door was bolted. Removes the force parameter, zombies and xenos now rely upon ToolForcePoweredComponent in order to be able to open powered doors.
## Media

https://github.com/space-wizards/space-station-14/assets/62370007/819461d9-8146-464c-b4c7-b5e298faab1e

https://github.com/space-wizards/space-station-14/assets/62370007/0dad30c5-cb96-46f0-b8fd-37c290a2b41c

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Zombies and xenos can no longer pry open bolted doors.

### CODE REVIEW

**Review by Partmedia — APPROVED** [review]
Thank you very much for your investigation and fix. I've reviewed and tested this in-game. Just waiting for a quick OK from the codeowners @EmoGarbage404 or @ElectroJr to merge.

File: Content.Server/Zombies/ZombieSystem.Transform.cs
Comment (keronshb) [inline Content.Server/Zombies/ZombieSystem.Transform.cs:172]: They shouldn't be relying on anything Tool related.

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml
Comment (keronshb) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:30]: They shouldn't be relying on anything Tool related.

**Review by keronshb — CHANGES_REQUESTED** [review]
The Tool Force solution isn't the ideal solution because that's for tools. Also I'm sure it might be intended for Xenos and Zombies to pry locked doors, but I could be wrong.

File: Content.Server/Zombies/ZombieSystem.Transform.cs
Comment (Partmedia) [reply Content.Server/Zombies/ZombieSystem.Transform.cs:172]: As far as I can tell, this component is a misnomer. It was introduced by #10770 for the jaws of life specifically to allow prying of powered doors. It is not used anywhere else.

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml
Comment (Partmedia) [reply Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:30]: See above.

File: Content.Server/Zombies/ZombieSystem.Transform.cs
Comment (keronshb) [reply Content.Server/Zombies/ZombieSystem.Transform.cs:172]: If that's the case then it should be renamed with a different comment.

**Comment by keronshb** [comment]
This should wait for #19394 to be merged first

**Review by keronshb — CHANGES_REQUESTED** [review]
Instead of doing it this way, there should be a bool on the comps to allow/disallow prying bolted doors. Xenos should be able to pry bolted doors where Zombies shouldn't.

**Comment by nikthechampiongr** [comment]
> This should wait for #19394 to be merged first

> Instead of doing it this way, there should be a bool on the comps to allow/disallow prying bolted doors.

I'll make sure to add a PryBolted data field for the PryingComponent to my PR tomorrow for this case then. It'll be set to true for zombies to maintain current behavior but if this pr is to be merged then this would probably fit in nicely.

**Comment by MisterMecky** [comment]
> > This should wait for #19394 to be merged first
> 
> > Instead of doing it this way, there should be a bool on the comps to allow/disallow prying bolted doors.
> 
> I'll make sure to add a PryBolted data field for the PryingComponent to my PR tomorrow for this case then. It'll be set to true for zombies to maintain current behavior but if this pr is to be merged then this would probably fit in nicely.

I didn't realize there was a prying factor in the works, I'm gonna close this as its probably better to be implemented in that PR.

**Pull request closed without merging**
